src/invidious/database/migrations/0008_create_playlists_table.cr

```diff
@@ -0,0 +1,48 @@
+module Invidious::Database::Migrations
+  class CreatePlaylistsTable < Migration
+    version 8
+
+    def up(conn : DB::Connection)
+      if !privacy_type_exists?(conn)
+        conn.exec <<-SQL
+        CREATE TYPE public.privacy AS ENUM
+        (
+          'Public',
+          'Unlisted',
+          'Private'
+        );
+        SQL
+      end
+
+      conn.exec <<-SQL
+      CREATE TABLE IF NOT EXISTS public.playlists
+      (
+        title text,
+        id text primary key,
+        author text,
+        description text,
+        video_count integer,
+        created timestamptz,
+        updated timestamptz,
+        privacy privacy,
+        index int8[]
+      );
+      SQL
+
+      conn.exec <<-SQL
+      GRANT ALL ON public.playlists TO current_user;
+      SQL
+    end
+
+    private def privacy_type_exists?(conn : DB::Connection) : Bool
+      !conn.query_one?(<<-SQL, as: Int32).nil?
+      SELECT 1 AS one
+      FROM pg_type
+      INNER JOIN pg_namespace ON pg_namespace.oid = pg_type.typnamespace
+      WHERE pg_namespace.nspname = 'public'
+        AND pg_type.typname = 'privacy'
+      LIMIT 1;
+      SQL
```

Comment:
As a matter of style/readibility, a separate request variable (like in the other DB functions) would be nice!